src/js/editor/editor.js

```diff
@@ -307,20 +299,27 @@ class Editor {
 
   handleDeletion(event) {
     event.preventDefault();
+    const key = Key.fromEvent(event),
+          direction = key.direction;
 
     let offsets = this.cursor.offsets;
     let currentMarker, currentOffset;
-    this.run((postEditor) => {
+
+    this.run(postEditor => {
       let results;
       if (this.cursor.hasSelection()) {
         results = postEditor.deleteRange(offsets);
       } else {
-        // FIXME: perhaps this should accept this.cursor.offsets?
-        results = postEditor.deleteCharAt(offsets.headMarker, offsets.headOffset-1);
+        const deletePosition = {
+          marker: offsets.headMarker,
+          offset: offsets.headOffset
+        };
+        results = postEditor.deleteFrom(deletePosition, direction);
```

Comment:
These all seem like different views of deleting a range to me. Backward 
should ideally be as simple as subtracting a position from the offsets.

As for the unified logic, I don’t think that should be take to mean 
“use one big function”. Instead I think it means to build for one 
master format of inputs. The semantic uses are then wrappers that adapt 
to that master format, and the internals of that format we then know to 
be pretty safe for all cases. The alternative is that you’re 
implementing the same deletion logic several times, and likely not 
covering all opportunities for failure.

Ideally the cursor doesn’t play into the data model changes. I’m 
aware it does right now, in that a currentMarker and offset are 
returned.